apps/frontend/src/pages/project/libs/components/project-menu/project-menu.tsx

```diff
@@ -0,0 +1,56 @@
+import { Menu, MenuItem, NavLink } from "~/libs/components/components.js";
+import { AppRoute, PermissionKey } from "~/libs/enums/enums.js";
+import { checkHasPermission } from "~/libs/helpers/helpers.js";
+import { useAppSelector, useCallback, usePopover } from "~/libs/hooks/hooks.js";
+
+type Properties = {
+	onEdit: () => void;
+	onManageAccess: () => void;
+};
+
+const ProjectMenu = ({
+	onEdit,
+	onManageAccess,
+}: Properties): JSX.Element | null => {
+	const { isOpened, onClose, onOpen } = usePopover();
+	const { authenticatedUser } = useAppSelector(({ auth }) => auth);
+
+	const handleEditClick = useCallback(() => {
+		onEdit();
+		onClose();
+	}, [onEdit, onClose]);
+
+	const handleManageAccessClick = useCallback(() => {
+		// No need to check for onManageAccess
+		onManageAccess();
+		onClose();
+	}, [onManageAccess, onClose]);
+
+	if (!authenticatedUser) {
+		return null;
+	}
+
+	const hasManageAllProjectsPermission = checkHasPermission(
+		[PermissionKey.MANAGE_ALL_PROJECTS],
+		authenticatedUser.groups.flatMap((group) => group.permissions),
+	);
+
+	if (!hasManageAllProjectsPermission) {
+		return null;
+	}
+
+	return (
+		<Menu isOpened={isOpened} onClose={onClose} onOpen={onOpen}>
+			<MenuItem iconName="pencil" label="Edit" onClick={handleEditClick} />
+			<NavLink className="menu-item" to={AppRoute.ACCESS_MANAGEMENT}>
+				<MenuItem
+					iconName="access"
+					label="Manage Access"
+					onClick={handleManageAccessClick}
+				/>
+			</NavLink>
```

Comment:
Does it work as we need with `NavLink` component and `onClick` in inside component? Won't the actions be triggered at the same time and cause a conflict?

Reply:
I split them into two useEffects to keep the logic cleaner and easier to follow. One handles fetching the project, and the other manages the modal state.

Reply:
I made sure we’re not using onClick with NavLink to avoid conflicts. Now we handle navigation and actions separately.